WooCommerce/Classes/ViewRelated/Products/Media/ProductImagesViewController.swift

```diff
@@ -208,6 +238,11 @@ private extension ProductImagesViewController {
         hasDeletedAnyImages = true
         productImageActionHandler.deleteProductImage(productImage)
     }
+
+    func handleProductImageStatusesReordering(_ productImageStatuses: [ProductImageStatus]) {
+        hasMovedAnyImages = true
+        productImageActionHandler.updateProductImageStatusesAfterReordering(productImageStatuses)
```

Comment:
To maintain the current flow, we do not update the statuses directly but forward the event to the action handler, and this handler will notify us when the statuses have been updated.